sbi/samplers/score/diffuser.py

```diff
@@ -99,11 +99,11 @@ def initialize(self, num_samples: int) -> Tensor:
         # batched sampling setting with a flag.
         # TODO: this fixes the iid setting shape problems, but iid inference via
         # iid_bridge is not accurate.
-        # num_batch = self.batch_shape.numel()
-        # init_shape = (num_batch, num_samples) + self.input_shape
-        init_shape = (
-            num_samples,
-        ) + self.input_shape  # just use num_samples, not num_batch
+        num_batch = self.batch_shape.numel()
+        init_shape = (num_samples, num_batch) + self.input_shape
+        # init_shape = (
+        #     num_samples,
+        # ) + self.input_shape  # just use num_samples, not num_batch
```

Comment:
what does this change implicate? aren't we still raising an error for the iid setting and fixing it in the other PR by Manuel?

Reply:
As far as I can tell, the former just matches our normal shape conventions, where as the (now commented out) version does not. But if there is some reason to keep the lines that are now commented out, we can revert this. @manuelgloeckler do you have any thoughts on this?

Reply:
No, changes do make sense, and the comments can be removed.